### PR TITLE
[MM-37064] - Onboarding flow is showing for existing users

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
All new users created from now on will have those preferences set to show for them, such that they can view the onboarding process the first time they log in or use the app.

#### Ticket Link
[MM-37064](https://mattermost.atlassian.net/browse/MM-37064?atlOrigin=eyJpIjoiZDZkNTg3N2E3OGFiNGNiZjg5OWZhMjE1YTgxZGJmZDkiLCJwIjoiaiJ9)

#### Release Note

```release-note
NONE

```

### CODE REVIEW

File: model/preference.go [hunk 1]
```diff
@@ -13,6 +13,24 @@ import (
...
+	PREFERENCE_CATEGORY_DISPLAY_SETTINGS      = "display_settings"
+	PREFERENCE_NAME_COLLAPSED_THREADS_ENABLED = "collapsed_reply_threads"
+	PREFERENCE_NAME_CHANNEL_DISPLAY_MODE      = "channel_display_mode"
+	PREFERENCE_NAME_COLLAPSE_SETTING          = "collapse_previews"
+	PREFERENCE_NAME_MESSAGE_DISPLAY           = "message_display"
+	PREFERENCE_NAME_NAME_FORMAT               = "name_format"
+	PREFERENCE_NAME_USE_MILITARY_TIME         = "use_military_time"
+	PREFERENCE_RECOMMENDED_NEXT_STEPS         = "recommended_next_steps"
+
+	PREFERENCE_CATEGORY_THEME           = "theme"
```
Comment (marianunez) [inline model/preference.go:33]: Are all these new preference const related?

Also, can we keep the same camel case for consistency?

**Review by marianunez — CHANGES_REQUESTED** [review]
Can we add a test to check the preferences are set correctly at users_test.go?

File: model/preference.go (hunk 1 above)
Comment (AGMETEOR) [reply model/preference.go:33]: I only added `PREFERENCE_RECOMMENDED_NEXT_STEPS ` but not sure why github is showing all those as additions I made.

Comment (AGMETEOR) [reply model/preference.go]: The issue was the way I resolved conflicts. Fixed now.

**Review by marianunez — APPROVED** [review]
Thanks @AGMETEOR!

**Review by nickmisasi — APPROVED** [review]

**Comment by AGMETEOR** [comment]
@stevemudie this will be tested as part of the corresponding webapp PR here https://github.com/mattermost/mattermost-webapp/pull/8442

**Comment by amyblais** [comment]
@AGMETEOR Needs a manual cherry pick

**Comment by AGMETEOR** [comment]
> @AGMETEOR Needs a manual cherry pick

On it

**Pull request merged**